### PR TITLE
fix: feature flag can edit 

### PR DESCRIPTION
**Pull request by liyiy**
## Problem
An issue occurred where standard cloud users were blocked from being able to edit feature flags because of the implementation of "can edit" for feature flags. This was because we were filtering for organization resource access objects directly without making it organization specific. We also needed to add in a quick return value of "true" for users who are not on the enterprise plan at all and therefore should be completely unaffected by feature flags permissions block.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/models/feature_flag.py [hunk 1]
```diff
@@ -743,7 +744,7 @@ def can_user_edit_feature_flag(request, feature_flag):
         all_role_memberships = request.user.role_memberships.select_related("role").all()
         try:
             feature_flag_resource_access = OrganizationResourceAccess.objects.get(
-                resource=OrganizationResourceAccess.Resources.FEATURE_FLAGS
+                organization=request.user.organization, resource=OrganizationResourceAccess.Resources.FEATURE_FLAGS
```
Comment (neilkakkar) [inline posthog/models/feature_flag.py:747]: same issue here I think: https://github.com/PostHog/posthog/blob/master/ee/api/feature_flag_role_access.py#L20

File: posthog/models/feature_flag.py [hunk 2]
```diff
@@ -727,6 +727,7 @@ def get_user_blast_radius(feature_flag_condition: dict, team: Team):
 
 
 def can_user_edit_feature_flag(request, feature_flag):
+    # self hosted check for enterprise models that may not exist
```
[the remark this answers is not in the document]
Comment (liyiy) [reply posthog/models/feature_flag.py:730]: I swore we used to have a method we would use called `has_available_feature` but as of currently that doesn't seem to exist anymore 🤷‍♀️

[the remark this answers is not in the document]
Comment (liyiy) [reply posthog/models/feature_flag.py:730]: I'm not entirely sure what the best way to grab that value is anymore and so I was using "get_available_features" for it but that seems to bump queries massively..

[the remark this answers is not in the document]
Comment (neilkakkar) [reply posthog/models/feature_flag.py:730]: Something like this?

https://github.com/PostHog/posthog/blob/master/posthog/api/event_definition.py#L100

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**